### PR TITLE
Implement Framebuffer Subpass support

### PR DESCRIPTION
**Pull request by reduz**
* Required for better optimizing mobile platforms
* Will be used by the Vulkan mobile renderer.
* Everything properly exposed in RenderingDevice

RenderingDevice changes:

image

image

### CODE REVIEW

**Review by BastiaanOlij — APPROVED** [review]
Coding style wise this looks good, also like the API here, looking forward to actually trying it out to see if it works properly.

**Comment by akien-mga** [comment]
Thanks!

File: drivers/vulkan/rendering_device_vulkan.cpp [hunk 1]
```diff
@@ -3498,17 +3648,29 @@ VkRenderPass RenderingDeviceVulkan::_render_pass_create(const Vector<AttachmentF
...
-		*r_color_attachment_count = color_references.size();
-	}
 	return render_pass;
 }
 
 RenderingDevice::FramebufferFormatID RenderingDeviceVulkan::framebuffer_format_create(const Vector<AttachmentFormat> &p_format, uint32_t p_view_count) {
+	FramebufferPass pass;
+	for (int i = 0; i < p_format.size(); i++) {
+		if (p_format[i].usage_flags & TEXTURE_USAGE_DEPTH_STENCIL_ATTACHMENT_BIT) {
+			pass.depth_attachment = i;
```
Comment (JFonS) [inline drivers/vulkan/rendering_device_vulkan.cpp:3658]: Should we check that only one depth attachment is passed?

File: doc/classes/RDFramebufferPass.xml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="RDFramebufferPass" inherits="RefCounted" version="4.0">
+	<brief_description>
+		Framebuffer pass attachment description.
+	</brief_description>
+	<description>
+		This class contains the list of attachment descriptions for a framebuffer pass. Each points with an index to a previously supplied list of texture attachments.
+	Multipass framebuffers can optimize some configurations in mobile, on desktop they provide little to no advantage.
```
Comment (JFonS) [inline doc/classes/RDFramebufferPass.xml:8]: I would mention TBDR somewhere here as the reason for optimization, while it's mostly used in mobile it's a better starting point for anyone trying to understand why this is useful. Soemthing like `Multipass framebuffers can optimize some configurations in Tile Based Deferred Rendering GPUs (moslty mobile, on desktop they provide little to no advantage).`

File: drivers/vulkan/rendering_device_vulkan.cpp (hunk 1 above)
Comment (reduz) [reply drivers/vulkan/rendering_device_vulkan.cpp:3658]: Technically you can pass as many as you want, but only one can be active per pass, but yeah I can add an error check if more than one is passed in the simple form.

**Pull request merged**